crates/swc_ecma_lints/src/rules/utils.rs

```diff
@@ -41,3 +42,51 @@ pub fn resolve_string_quote_type(source_map: &Arc<SourceMap>, lit_str: &Str) ->
         _ => None,
     }
 }
+
+#[derive(Debug)]
+pub enum ArgValue {
+    Str(JsWord),
+    Number(f64),
+    RegExp { exp: JsWord, flags: JsWord },
+    Ident,
+    Other,
+}
+
+pub fn extract_arg_val(expr: &Expr, check_parens: bool) -> ArgValue {
+    match expr {
+        Expr::Ident(_) => ArgValue::Ident,
+        Expr::Paren(ParenExpr { expr, .. }) => {
+            if check_parens {
+                extract_arg_val(expr.as_ref(), check_parens)
+            } else {
+                ArgValue::Other
+            }
+        }
+        Expr::Lit(Lit::Str(Str { value, .. })) => ArgValue::Str(value.clone()),
+        Expr::Lit(Lit::Num(Number { value, .. })) => ArgValue::Number(*value),
+        Expr::Lit(Lit::Regex(Regex { exp, flags, .. })) => ArgValue::RegExp {
+            exp: exp.clone(),
+            flags: flags.clone(),
+        },
+        Expr::Tpl(Tpl { exprs, .. }) => {
+            if exprs.is_empty() {
+                // TODO: extract text from tpl
+                ArgValue::Str(js_word!(""))
```

Comment:
This seems wrong to me

Reply:
I can read value from source map

Reply:
I mean, this will  make `parseInt(10, `foo`)`  treated as `parseInt(10, '')`

Reply:
Actually it's no matter for `radix` rule for now. Tpl handling required for rule `prefer_regex_literals`

Reply:
Oh... Seems like missed it while reviewing the PR for regex literals rule